src/shared/utilities/cliUtils.ts

```diff
@@ -0,0 +1,455 @@
+import { ext } from '../extensionGlobals'
+import { getIdeProperties } from '../extensionUtilities'
+import { makeTemporaryToolkitFolder, tryRemoveFolder } from '../filesystemUtilities'
+import { getLogger } from '../logger'
+import { HttpResourceFetcher } from '../resourcefetcher/httpResourceFetcher'
+import * as telemetry from '../telemetry/telemetry'
+import { ChildProcess } from '../utilities/childProcess'
+import { Window } from '../vscode/window'
+
+import * as nls from 'vscode-nls'
+import { Timeout } from './timeoutUtils'
+import { showMessageWithCancel } from './messages'
+const localize = nls.loadMessageBundle()
+
+const msgDownloading = localize('AWS.installProgress.downloading', 'downloading...')
+const msgInstallingLocal = localize('AWS.installProgress.installingLocal', 'installing local copy...')
+
+export class InstallerError extends Error {}
+export class InvalidPlatformError extends Error {}
+
+interface Cli {
+    command: {
+        unix: string
+        windows: string
+    }
+    source: {
+        macos: string
+        windows: string
+        linux: string
+    }
+    manualInstallLink: string
+    name: string
+}
+
+type AwsClis = 'aws' | 'ssm'
+
+/**
+ * CLIs and their full filenames and download paths for their respective OSes
+ * TODO: Add SAM? Other CLIs?
+ */
+export const AWS_CLIS: { [cli in AwsClis]: Cli } = {
+    aws: {
+        command: {
+            unix: path.join('AWSCLIV2', 'aws'),
+            windows: path.join('AWSCLIV2', 'aws.exe'),
+        },
+        source: {
+            macos: 'https://awscli.amazonaws.com/AWSCLIV2.pkg',
+            windows: 'https://awscli.amazonaws.com/AWSCLIV2.msi',
+            linux: 'https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip',
+        },
+        name: 'AWS',
+        manualInstallLink: 'https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html',
+    },
+    ssm: {
+        command: {
+            unix: path.join('sessionmanagerplugin', 'bin', 'session-manager-plugin'),
+            windows: path.join('sessionmanagerplugin', 'bin', 'session-manager-plugin.exe'),
+        },
+        source: {
+            // use pkg: zip is unsigned
+            macos: 'https://s3.amazonaws.com/session-manager-downloads/plugin/latest/mac/session-manager-plugin.pkg',
+            windows:
+                'https://session-manager-downloads.s3.amazonaws.com/plugin/latest/windows/SessionManagerPlugin.zip',
+            linux: 'https://s3.amazonaws.com/session-manager-downloads/plugin/latest/ubuntu_64bit/session-manager-plugin.deb',
+        },
+        name: 'Session Manager Plugin',
+        manualInstallLink:
+            'https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html',
+    },
+}
+
+/**
+ * Installs a selected CLI: wraps confirmation, cleanup, and telemetry logic.
+ * @param cli CLI to install
+ * @param confirmBefore Prompt before starting install?
+ * @returns Dir containing CLI
```

Comment:
Path to the binary would be better than the directory. It's easy to get the directory from the binary, but the inverse is not true.